pkg/controller/gitrepo/gitrepo_reconcile.go

```diff
@@ -145,6 +145,8 @@ func (r *ReconcileGitRepo) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 		phase := synv1alpha1.Created
```

Comment:
These few lines are exactly the same as for the creation case (L120-122). It would make sense to move that in a function.

Reply:
Was there a reason we `return`ed in the case the repo was newly created?

I removed the return which makes the logic easier.

Reply:
Yes makes sense. I can't remember a specific reason. It's probably something from before `handleRepoError` was introduced.